src/binary-reader.h

```diff
@@ -82,11 +82,12 @@ class BinaryReaderDelegate {
   /* Type section */
   virtual Result BeginTypeSection(Offset size) = 0;
   virtual Result OnTypeCount(Index count) = 0;
-  virtual Result OnType(Index index,
-                        Index param_count,
-                        Type* param_types,
-                        Index result_count,
-                        Type* result_types) = 0;
+  virtual Result OnFuncType(Index index,
+                            Index param_count,
+                            Type* param_types,
+                            Index result_count,
+                            Type* result_types) = 0;
+  virtual Result OnStructType(Index index) = 0;  // TODO
```

Comment:
Not sure what the TODO here is for?

Reply:
Oh yeah, this was a TODO to add information about the struct fields to the callback. I'll remove it in the next PR. :-)